refactor(shopware): Optimize `getPlugins` to use json output

mralexandernickel opened the pull request:
Refactoring the `getPlugins` function to not parse regular commandline output anymore, instead use the --json flag `plugin:list`. Refactoring 'sw:plugin:update:all' to not upgrade every installed plugin, but every installed plugin that in fact is upgradable.

- [ ] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Tests added?
- [x] Docs added?

      Please, regenerate docs by running next command:
      $ php bin/docgen

Schrank commented on recipe/shopware.php:
```diff
@@ -105,42 +105,18 @@
-        $pluginInformation = $splitRowIntoStructure($row);
-        $plugins[] = $pluginInformation;
-    }
+    $output = explode("\n", run('cd {{release_path}} && {{bin/console}} plugin:list --json'));
```
Maybe I'm not awake yet, but explode returns an array and json_decode expectes a string? 🤔

mralexandernickel replied:
@Schrank probably copy&paste error 🤦...give me some minutes, will verify and probably fix  😅

Schrank commented:
Thank you so much for taking care of that. It is on my list for months.

Schrank commented:
And please regenerate the doc:

Please, regenerate docs by running next command:
$ php bin/docgen

Thanks! :)

mralexandernickel commented:
@Schrank so, I've removed the explode (obviously^^), and used these changes within a real projects deployment, looking good to me. And I've also updated the docs 😊

And thank you very much for your effort on this project! ❤

Schrank commented:
Awesome, thank you!
